StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs

```diff
@@ -91,14 +98,20 @@ protected override void HandleXmlElement(SyntaxNodeAnalysisContext context, bool
                             {
                                 if (!textWithoutTrailingWhitespace.EndsWith(".", StringComparison.Ordinal))
                                 {
-                                    var location = Location.Create(xmlElement.SyntaxTree, new TextSpan(textToken.SpanStart + textWithoutTrailingWhitespace.Length, 1));
+                                    var location = Location.Create(xmlElement.SyntaxTree, new TextSpan(reportingLocation ?? textToken.SpanStart + textWithoutTrailingWhitespace.Length, 1));
                                     context.ReportDiagnostic(Diagnostic.Create(Descriptor, location));
                                 }
 
                                 elementDone = true;
                             }
                         }
                     }
+                    else if (xmlElement.Content[i].IsInlineElement())
+                    {
+                        // If a diagnostic gets reported for the element, place the diagnostic after the last inline
+                        // element.
+                        reportingLocation = xmlElement.Content[i].Span.End;
```

Comment:
This will fail for 2 edge cases:
- ```<summary>Text <see cref="x"/><see cref="y"/></summary>``` will report the wrong location.
- ```<summary>Test text. <see cref="x"/></summary>``` will not report anything and it should.

I think that when an inline element is found, there will not be a period, so a diagnostic can be reported straight away and the loop can be ended. This should solve both edge cases.

Reply:
I meant to write this:

```
reportingLocation = reportingLocation ?? ...;
```

That explains the first issue. I need to think more about the handling of empty elements because it's possible for other situations to occur with non-empty elements.

Reply:
@vweijsters I fixed the first case in ecee2c7e206890e27fb73d7bffc0c5938773d6e5 and then fixed the second case in bf6fec59f8f5e7f2692a12fc0ef5f55fb644a38d.